### PR TITLE
Feature/hdr histogram c

### PR DESCRIPTION
**Pull request by tarc**
Specify library name and version:  **hdrhistogram-c/0.11.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/hdrhistogram-c/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    license = "BSD 2-Clause License"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/HdrHistogram/HdrHistogram_c"
+    description = "'C' port of High Dynamic Range (HDR) Histogram"
+    topics = ("libraries", "c", "histogram")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"fPIC": [True, False],
+               "shared": [True, False]}
+    default_options = {"fPIC": True,
+                       "shared": False}
```
Comment (uilianries) [inline recipes/hdrhistogram-c/all/conanfile.py:17]:
```suggestion
                       "shared": False}
    requires = "zlib/1.2.11"
```

File: recipes/hdrhistogram-c/all/patches/0001-add-conan-support.patch [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ 
+diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
+index 9dbb311..5bd6ca4 100644
+--- a/src/CMakeLists.txt
++++ b/src/CMakeLists.txt
+@@ -1,10 +1,3 @@
+-find_package(ZLIB REQUIRED)
+-find_package(Threads REQUIRED)
+-
+-include(CheckLibraryExists)
```
Comment (uilianries) [inline recipes/hdrhistogram-c/all/patches/0001-add-conan-support.patch:25]: It should work normal without this patch. All system libs should listed according in cpp_info.system_libs

Comment (tarc) [reply recipes/hdrhistogram-c/all/patches/0001-add-conan-support.patch]: Done, cf. 2fa63c9

File: recipes/hdrhistogram-c/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class QuickfixConan(ConanFile):
+    name = "hdrhistogram-c"
+    license = "BSD 2-Clause License"
```
Comment (uilianries) [inline recipes/hdrhistogram-c/all/conanfile.py:8]:
```suggestion
    license = ("BSD-2-Clause", "CC0-1.0")
```

Short term version: https://spdx.org/licenses/BSD-2-Clause.html

File: recipes/hdrhistogram-c/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            if self.options.shared:
+                self._cmake.definitions["HDR_HISTOGRAM_BUILD_SHARED"] = True
+                self._cmake.definitions["HDR_HISTOGRAM_INSTALL_SHARED"] = True
+                self._cmake.definitions["HDR_HISTOGRAM_BUILD_STATIC"] = False
+                self._cmake.definitions["HDR_HISTOGRAM_INSTALL_STATIC"] = False
+            else:
+                self._cmake.definitions["HDR_HISTOGRAM_BUILD_SHARED"] = False
+                self._cmake.definitions["HDR_HISTOGRAM_INSTALL_SHARED"] = False
+                self._cmake.definitions["HDR_HISTOGRAM_BUILD_STATIC"] = True
+                self._cmake.definitions["HDR_HISTOGRAM_INSTALL_STATIC"] = True
```
Comment (uilianries) [inline recipes/hdrhistogram-c/all/conanfile.py:44]:
```suggestion
            self._cmake.definitions["HDR_HISTOGRAM_BUILD_SHARED"] = self.options.shared
            self._cmake.definitions["HDR_HISTOGRAM_INSTALL_SHARED"] = self.options.shared
            self._cmake.definitions["HDR_HISTOGRAM_BUILD_STATIC"] = not self.options.shared
            self._cmake.definitions["HDR_HISTOGRAM_INSTALL_STATIC"] = not self.options.shared
```

File: recipes/hdrhistogram-c/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+            self._cmake.configure(source_folder=self._source_subfolder, build_folder=self._build_subfolder)
+        return self._cmake
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("HdrHistogram_c-" + self.version, self._source_subfolder)
+
+    def requirements(self):
+        pass
+
```
Comment (uilianries) [inline recipes/hdrhistogram-c/all/conanfile.py:54]:
```suggestion
```

Unused

File: recipes/hdrhistogram-c/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
```
Comment (uilianries) [inline recipes/hdrhistogram-c/all/conanfile.py:64]:
```suggestion
        for patch in self.conan_data.get("patches", {}).get(self.version, []):
```

File: recipes/hdrhistogram-c/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+        self.copy("LICENSE.txt", dst="licenses", src=self._source_subfolder)
```
Comment (uilianries) [inline recipes/hdrhistogram-c/all/conanfile.py:72]:
```suggestion
        self.copy("LICENSE.txt", dst="licenses", src=self._source_subfolder)
        self.copy("COPYING.txt", dst="licenses", src=self._source_subfolder)
```

File: recipes/hdrhistogram-c/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+        if self.settings.os == "Linux":
+            self.cpp_info.system_libs = ["m"]
+        self.cpp_info.includedirs = [os.path.join("include", "hdr")]
```
Comment (uilianries) [inline recipes/hdrhistogram-c/all/conanfile.py:79]:
```suggestion
        self.cpp_info.includedirs = ["include", os.path.join("include", "hdr")]
```

File: recipes/hdrhistogram-c/all/conanfile.py [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+        self.copy("LICENSE.txt", dst="licenses", src=self._source_subfolder)
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+        if self.settings.os == "Linux":
+            self.cpp_info.system_libs = ["m"]
```
Comment (uilianries) [inline recipes/hdrhistogram-c/all/conanfile.py:78]:
```suggestion
            self.cpp_info.system_libs = ["m", "rt"]
```

File: recipes/hdrhistogram-c/all/conanfile.py [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+        self.cpp_info.includedirs = [os.path.join("include", "hdr")]
+
+        if self.settings.os == "Windows":
```
Comment (uilianries) [inline recipes/hdrhistogram-c/all/conanfile.py:81]:
```suggestion
        elif self.settings.os == "Windows":
```

File: recipes/hdrhistogram-c/all/conanfile.py [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+            self.cpp_info.system_libs.extend(["ws2_32"])
+        elif self.settings.os == "Linux":
+            self.cpp_info.system_libs.extend(["rt"])
+            self.cpp_info.system_libs.extend(["m"])
```
Comment (uilianries) [inline recipes/hdrhistogram-c/all/conanfile.py:85]:
```suggestion
```

Repeated

File: recipes/hdrhistogram-c/all/patches/0001-add-conan-support.patch [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+-        PRIVATE
+-            ZLIB::ZLIB
+-            Threads::Threads
+-            $<$<BOOL:${HAVE_LIBM}>:m>
+-            $<$<BOOL:${HAVE_LIBRT}>:rt>
+-            $<$<BOOL:${WIN32}>:ws2_32>)
+-    target_include_directories(${NAME}
+-        PUBLIC
+-            $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+-            $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
```
Comment (uilianries) [inline recipes/hdrhistogram-c/all/patches/0001-add-conan-support.patch:35]:
```suggestion
-            ZLIB::ZLIB
-            Threads::Threads
-            $<$<BOOL:${HAVE_LIBM}>:m>
-            $<$<BOOL:${HAVE_LIBRT}>:rt>
-            $<$<BOOL:${WIN32}>:ws2_32>)
-    target_include_directories(${NAME}
-        PUBLIC
-            $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
-            $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
```

cmake_find_package generator should be enough. This patch can be removed.

File: recipes/hdrhistogram-c/all/test_package/conanfile.py [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+    generators = "cmake"
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self.settings):
+            bin_path = os.path.join("bin", "hdr_decoder_c")
```
Comment (uilianries) [inline recipes/hdrhistogram-c/all/test_package/conanfile.py:16]: Only one example should be enough.

File: recipes/hdrhistogram-c/all/patches/0001-add-conan-support.patch (hunk 12 above)
Comment (tarc) [reply recipes/hdrhistogram-c/all/patches/0001-add-conan-support.patch]: I didn't know of `cmake_find_package`! Great!

Done, cf. 6fc4a7a9

File: recipes/hdrhistogram-c/all/test_package/conanfile.py (hunk 13 above)
Comment (tarc) [reply recipes/hdrhistogram-c/all/test_package/conanfile.py]: Done, cf. 24d1f956

**Pull request merged**